internal/internal_nexus_task_handler.go

```diff
@@ -0,0 +1,353 @@
+		timeout, err := time.ParseDuration(timeoutStr)
+		if err != nil {
+			return nil, nil, nexusHandlerError(nexus.HandlerErrorTypeBadRequest, "cannot parse request timeout")
+		}
+		ctx, cancel := context.WithTimeout(ctx, timeout)
+		return ctx, cancel, nil
+	}
+
+	return ctx, func() {}, nil
+}
+
+func (h *nexusTaskHandler) loggerForTask(response *workflowservice.PollNexusTaskQueueResponse) (log.Logger, *nexuspb.HandlerError) {
+	var service, operation string
+
+	switch req := response.GetRequest().GetVariant().(type) {
+	case *nexuspb.Request_StartOperation:
+		service = req.StartOperation.Service
+		operation = req.StartOperation.Operation
+	case *nexuspb.Request_CancelOperation:
+		service = req.CancelOperation.Service
+		operation = req.CancelOperation.Operation
+	default:
+		return nil, nexusHandlerError(nexus.HandlerErrorTypeNotImplemented, "unknown request type")
+	}
+
+	return log.With(h.logger,
+		tagNexusService, service,
+		tagNexusOperation, operation,
+	), nil
+}
+
+func (h *nexusTaskHandler) metricsHandlerForTask(response *workflowservice.PollNexusTaskQueueResponse) (metrics.Handler, *nexuspb.HandlerError) {
+	var service, operation string
+
+	switch req := response.GetRequest().GetVariant().(type) {
+	case *nexuspb.Request_StartOperation:
+		service = req.StartOperation.Service
+		operation = req.StartOperation.Operation
+	case *nexuspb.Request_CancelOperation:
+		service = req.CancelOperation.Service
+		operation = req.CancelOperation.Operation
+	default:
+		return nil, &nexuspb.HandlerError{
+			ErrorType: string(nexus.HandlerErrorTypeNotImplemented),
+			Failure: &nexuspb.Failure{
+				Message: "unknown request type",
+			},
+		}
+	}
+
+	return h.metricsHandler.WithTags(metrics.NexusTags(service, operation, h.taskQueueName)), nil
+}
+
+func (h *nexusTaskHandler) fillInCompletion(taskToken []byte, res *nexuspb.Response) *workflowservice.RespondNexusTaskCompletedRequest {
+	r := proto.Clone(h.baseCompletion).(*workflowservice.RespondNexusTaskCompletedRequest)
+	r.TaskToken = taskToken
+	r.Response = res
+	return r
+}
+
+func (h *nexusTaskHandler) fillInFailure(taskToken []byte, err *nexuspb.HandlerError) *workflowservice.RespondNexusTaskFailedRequest {
+	r := proto.Clone(h.baseFailure).(*workflowservice.RespondNexusTaskFailedRequest)
+	r.TaskToken = taskToken
+	r.Error = err
+	return r
+}
+
+// payloadSerializer is a fake nexus Serializer that uses a data converter to read from an embedded payload instead of
+// using the given nexus.Context. Supports only Deserialize.
+type payloadSerializer struct {
+	converter converter.DataConverter
+	payload   *common.Payload
+}
+
+func (p *payloadSerializer) Deserialize(_ *nexus.Content, v any) error {
+	return p.converter.FromPayload(p.payload, v)
+}
+
+func (p *payloadSerializer) Serialize(v any) (*nexus.Content, error) {
+	panic("unimplemented") // not used - operation outputs are directly serialized to payload.
+}
+
+var nopReadCloser = io.NopCloser(bytes.NewReader([]byte{}))
```

Comment:
This is not a `nopReadCloser`, this `emptyReaderNopCloser`. This is going to return an EOF immediately (which is maybe what you want, though can change empty byte slice to `nil`)

Reply:
Sure, I can rename. It can't be nil because the Nexus SDK tries to consume it.

Reply:
I mean you can use `nil` instead of `[]byte{}`